hal/armv7a/_memcpy.S

```diff
@@ -0,0 +1,299 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system kernel
+ *
+ * hal_memcpy
+ *
+ * Copyright 2024 Phoenix Systems
+ * Author: Lukasz Leczkowski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#define DST r0
+#define SRC r1
+#define LEN r12
```

Comment:
If you use instead of the `r12 register some one in the `r0-r7` range you can reduce the size of the code.
Thus, for example, using `r5`
```
#define LEN r5
...
push {r0, r5}
...
pop {r0, r5}
bx lr
```
Result:
```
$ size _memcpy.o

LEN=r12 560
LEN=r5 488
```